### PR TITLE
Add windows auth tests for inproc

### PR DESCRIPTION
**Pull request by jkotalik**
Tests for #428

### CODE REVIEW

File: test/IISExpress.FunctionalTests/WindowsAuthTests.cs [hunk 1]
```diff
@@ -26,13 +26,14 @@ public static TestMatrix TestVariants
...
-                .WithAllAncmVersions();
+                .WithAllAncmVersions()
+                .WithAllHostingModels();
 
         [ConditionalTheory]
         [MemberData(nameof(TestVariants))]
         public async Task WindowsAuthTest(TestVariant variant)
         {
-            var deploymentParameters = _fixture.GetBaseDeploymentParameters(variant);
+            var deploymentParameters = _fixture.GetBaseDeploymentParameters(variant, publish: true);
```
Comment (jkotalik) [inline test/IISExpress.FunctionalTests/WindowsAuthTests.cs]: One of the inproc tests was failing without setting publish to true.

Comment (pakrym) [reply test/IISExpress.FunctionalTests/WindowsAuthTests.cs]: Why?

File: src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs [hunk 2]
```diff
@@ -293,7 +293,7 @@ private void PrepareConfig(string contentRoot, int port)
 
             ConfigureModuleAndBinding(config.Root, contentRoot, port);
 
-            if (!DeploymentParameters.PublishApplicationBeforeDeployment)
+            if (!DeploymentParameters.PublishApplicationBeforeDeployment && !File.Exists(Path.Combine(contentRoot, "web.config")))
```
Comment (pakrym) [inline src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs]: So we always publish any app that contains web.config?

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs]: What is the reason?

Comment (jkotalik) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs]: No. The reason we need this check is if a non-published app contains a web.config (e.g. the InProcessTestSite), we don't want to re-add the aspnetcore element. Before, all tests with the inprocess app were published; now they are not.

Comment (pakrym) [reply src/Microsoft.AspNetCore.Server.IntegrationTesting.IIS/IISExpressDeployer.cs]: I see, extract a var.

File: test/IISExpress.FunctionalTests/WindowsAuthTests.cs [hunk 3]
```diff
@@ -26,7 +26,7 @@ public static TestMatrix TestVariants
             => TestMatrix.ForServers(DeployerSelector.ServerType)
                 .WithTfms(Tfm.NetCoreApp22, Tfm.Net461)
                 .WithAllApplicationTypes()
-                .WithAllAncmVersions();
+                .WithAncmV2InProcess();
```
Comment (pakrym) [inline test/IISExpress.FunctionalTests/WindowsAuthTests.cs]: Why?

File: test/IISExpress.FunctionalTests/WindowsAuthTests.cs [hunk 4]
```diff
@@ -27,7 +27,7 @@ public static TestMatrix TestVariants
                 .WithTfms(Tfm.NetCoreApp22, Tfm.Net461)
                 .WithAllApplicationTypes()
                 .WithAllAncmVersions();
-
+
```
Comment (pakrym) [inline test/IISExpress.FunctionalTests/WindowsAuthTests.cs]: nit: newline

**Comment by pakrym** [comment]
LGTM, merge it after merging apps.

**Comment by jkotalik** [comment]
Test failure #1389

**Comment by jkotalik** [comment]
@dotnet-bot test this please

**Comment by pakrym** [comment]
Make sure tests are running for inproc.

File: test/IISExpress.FunctionalTests/WindowsAuthTests.cs [hunk 5]
```diff
@@ -26,7 +26,8 @@ public static TestMatrix TestVariants
             => TestMatrix.ForServers(DeployerSelector.ServerType)
                 .WithTfms(Tfm.NetCoreApp22, Tfm.Net461)
                 .WithAllApplicationTypes()
-                .WithAllAncmVersions();
+                .WithAllAncmVersions()
+                .WithAllHostingModels();
 
         [ConditionalTheory]
         [MemberData(nameof(TestVariants))]
```
Comment (Tratcher) [inline test/IISExpress.FunctionalTests/WindowsAuthTests.cs]: The actual test should confirm which response is expected based on which hosting model. It looks like in-proc would return "backcompat;..."

**Comment by Tratcher** [comment]
backcompat was for older ANCM versions that didn't set ASPNETCORE_IIS_HTTPAUTH. Not sure why that's still being tested.

**Comment by jkotalik** [comment]
I'll remove it from the test.

**Pull request merged**